pkg/util/tests/kubectl/kubectl.go

```diff
@@ -0,0 +1,304 @@
+		// Workaround because the resource can be already deleted
+		if strings.Contains(output, "not found") {
+			return nil
+		}
+
+		return fmt.Errorf("error deleting yaml: %v", err)
+
+	}
+
+	return nil
+}
+
+// GetCondition returns the condition of a resource
+func GetCondition(ns, resourceType, resourceName string) ([]r.Condition, error) {
+	var resource r.Resource
+
+	output, err := GetJSON(ns, resourceType, resourceName)
+	if err != nil {
+		return []r.Condition{}, err
+	}
+
+	err = json.Unmarshal([]byte(output), &resource)
+	if err != nil {
+		return []r.Condition{}, err
+	}
+
+	return resource.Status.Conditions, nil
+}
+
+// GetPodPhase returns the phase of a pod
+func GetPodPhase(ns, selector string) (string, error) {
+	var resource r.Resource
+
+	podName, err := GetPodName(ns, selector)
+	if err != nil {
+		return "", err
+	}
+
+	output, err := GetJSON(ns, "pod", podName)
+	if err != nil {
+		return "", err
+	}
+
+	err = json.Unmarshal([]byte(output), &resource)
+	if err != nil {
+		return "", err
+	}
+
+	return resource.Status.Phase, nil
+}
+
+// GetCRDs returns all the CRDs names in a list
+func GetCRDs() ([]string, error) {
+	cmd := kubectl("get crds -o jsonpath={.items[*].metadata.name}")
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return []string{}, fmt.Errorf("error getting crds: %v", err)
+	}
+
+	return strings.Split(output, " "), nil
+}
+
+// GetResourceList returns a json list of the resources of a namespace
+func GetResourceList(ns string) (r.ResourceList, error) {
+	// TODO: improve the function to get all the resources
+	output, err := GetJSON(ns, "all", "")
+	if err != nil {
+		return EmptyResourceList, err
+	}
+
+	var resourceList r.ResourceList
+	err = json.Unmarshal([]byte(output), &resourceList)
+	if err != nil {
+		return EmptyResourceList, err
+	}
+
+	// Return an empty list if there are no resources
+	if len(resourceList.Items) == 0 {
+		return EmptyResourceList, nil
+	}
+
+	return resourceList, nil
+}
+
+// GetJSON returns the json of a resource
+// Arguments:
+// - ns: namespace
+// - resourceType: type of the resource
+// - resourceName: name of the resource
+func GetJSON(ns, resourceType, resourceName string) (string, error) {
+	cmd := kubectl("get %s %s -n %s -o json", resourceType, resourceName, ns)
+	json, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	return json, nil
+}
+
+// GetPodName returns the pod name from a label, if there is more than one pod, it will return an error
+func GetPodName(ns, label string) (string, error) {
+	var podList []string
+	podList, err := GetPodsName(ns, label)
+	if err != nil {
+		return "", err
+	}
+	if len(podList) > 1 {
+		return "", fmt.Errorf("more than one pod found with label %s", label)
+	}
+	if len(podList) == 0 {
+		return "", fmt.Errorf("no pod found with label %s", label)
+	}
+
+	return podList[0], nil
+}
+
+// GetPodsName returns the pod name from a label
+func GetPodsName(ns, label string) ([]string, error) {
```

Comment:
This needs to be `GetPodNames`. But didn't we say that we'll just use `GetXyz` for functions that return names and use `GetXyzJSON` for functions that return JSON?
```suggestion
func GetPods(ns, label string) ([]string, error) {
```